action.yml

```diff
@@ -1,34 +1,77 @@
 name: Dispatch CI
 description: GitHub Action to trigger ci.yml workflow_dispatch event via GitHub API
 
+inputs:
+  major_type:
+    type: string
+    required: false
+    default: 'dynamic'
+  minor_type:
+    type: string
+    required: false
+    default: 'dynamic'
+
 runs:
   using: composite
   steps:
+    - name: Validate inputs
+      shell: bash
+      env:
+        INPUTS_MAJOR_TYPE: ${{ inputs.major_type }}
+        INPUTS_MINOR_TYPE: ${{ inputs.minor_type }}
+      run: |
+        if [[ $INPUTS_MAJOR_TYPE != "dynamic" ]] && \
+          [[ $INPUTS_MAJOR_TYPE != "current" ]] && \
+          [[ $INPUTS_MAJOR_TYPE != "next" ]] && \
+          [[ $INPUTS_MAJOR_TYPE != "previous" ]]
+        then
+          echo "Invalid input major_type"
+          exit 1
+        fi
+        if [[ $INPUTS_MINOR_TYPE != "dynamic" ]] && \
+          [[ $INPUTS_MINOR_TYPE != "next-minor" ]] && \
+          [[ $INPUTS_MINOR_TYPE != "next-patch" ]]
+        then
+          echo "Invalid input minor_type"
+          exit 1
+        fi
+
     - name: Derive types
       id: types
       shell: bash
       env:
+        INPUTS_MINOR_TYPE: ${{ inputs.minor_type }}
+        INPUTS_MAJOR_TYPE: ${{ inputs.major_type }}
         GITHUB_REPOSITORY: ${{ github.repository }}
       run: |
-        # Repository that has a twice-weekly schedule run on consecutive days - use day of year
+        # Repositories run on a three times weekly schedule run on consecutive days
+        # Get the current day of year (e.g. Feb 15 is 46) which will be used for a mod calc below
         DATE_PART=$(date +%-j)
         MAJOR_TYPE=""
         MINOR_TYPE=""
-        if [[ $GITHUB_REPOSITORY == "silverstripe/installer" ]] || \
-            [[ $GITHUB_REPOSITORY == "silverstripe/recipe-kitchen-sink" ]]
-        then
-          # Respository that has daily schedule - use hour
-          DATE_PART=$(date +%H)
-        fi
-        if (( $DATE_PART % 2 == 1 )); then
+        if (( $DATE_PART % 3 == 1 )); then
           # Current major, next-minor e.g. "6"
           MAJOR_TYPE="current"
           MINOR_TYPE="next-minor"
+        elif (( $DATE_PART % 3 == 2 )); then
+          # Current major, next-patch e.g. "6.0"
+          MAJOR_TYPE="current"
+          MINOR_TYPE="next-patch"
         else
           # Previous major, next-patch e.g. "5.4"
           MAJOR_TYPE="previous"
           MINOR_TYPE="next-patch"
         fi
+        echo "dynamic MAJOR_TYPE is $MAJOR_TYPE"
+        echo "dynamic MINOR_TYPE is $MINOR_TYPE"
+        if [[ "$INPUTS_MAJOR_TYPE" != 'dynamic' ]]; then
+          echo "Using input override $INPUTS_MAJOR_TYPE for MAJOR_TYPE"
+          MAJOR_TYPE="$INPUTS_MAJOR_TYPE"
+        fi
+        if [[ "$INPUTS_MINOR_TYPE" != 'dynamic' ]]; then
+          echo "Using input override $INPUTS_MINOR_TYPE for MINOR_TYPE"
+          MINOR_TYPE="$INPUTS_MINOR_TYPE"
+        fi
```

Comment:
Why are we bothering to get the dynamic types before even looking at the inputs?

If we have non-dynamic inputs, we should just output those to `$GITHUB_OUTPUT` at the top of this step and skip the dynamic stuff.

Reply:
Logic would get more convoluted if we do that because the two inputs can be independently set e.g. one dynamic, the other set to something like 'next'

Reply:
This is probably moot anyway given https://github.com/silverstripe/gha-dispatch-ci/pull/27#issuecomment-3161900884 so lets resolve that conversation first and circle back if we end up keeping the two inputs.